### DISCUSSION TITLE
An option to skip values matching defaults from `createSerializer(parsers)(values)` output

### DISCUSSION THREAD

**Discussion by ivan-kleshnin** [47ng/nuqs#570] [Ideas] [answered]
**Is there an option to skip values matching defaults from `createSerializer(parsers)(values)` output?**

So, e.g. `serializer(...)` here:

```ts
const parsers = {
  contacts: parseAsArrayOf(parseAsString).withDefault([]),
}

const serializer = createSerializer(parsers)
serializer(url, values) // !!!
```

would return `"?"` or an empty string instead of `"?contacts="`. It's easier to visually debug URL issues when URLs are short and in my project we may have tens of filters most of which would be keys representing the default case.

[Here's](https://github.com/47ng/nuqs/discussions/471) a similar issue but it ends up with accepting `null` value and avoiding defaults as a solution. 

I don't see it as an acceptable solution, I would like to avoid `null`-containing unions to keep TS types clean. 

I see no technical blocker that would make the goal of avoiding query params matching their defaults non-achievable.

**Comment by franky47** [comment]
It could be a matter of checking the value against the default (if any, and using the `eq` property of the parser to handle non-primitives like arrays), and skipping adding it to the output. This should probably be an option, as in some cases you might want explicit _current_ defaults, since those can change over the lifetime of the app.

Would you like to open a PR?

**Reply by franky47** [reply, marked as answer]
FYI, this is being worked on in #720 and should land in v2.1.0.
